oauth2_http/java/com/google/auth/oauth2/GoogleCredentials.java

```diff
@@ -251,19 +321,64 @@ public GoogleCredentials(AccessToken accessToken) {
     this(accessToken, null);
   }
 
+  /**
+   * Constructor with using builder All the fields comes explicitly from builder.
+   *
+   * @param builder an instance of a builder
+   */
   protected GoogleCredentials(Builder builder) {
-    this(builder.getAccessToken(), builder.getQuotaProjectId());
+    super(builder.getAccessToken());
+    this.quotaProjectId = builder.getQuotaProjectId();
+    this.universeDomain = builder.getUniverseDomain();
   }
 
   /**
-   * Constructor with explicit access token and refresh times
+   * Constructor with explicit access token and refresh times.
+   *
+   * <p>Deprecated, please use the builder constructor whenever possible.
    *
    * @param accessToken initial or temporary access token
    */
+  @Deprecated
   protected GoogleCredentials(
       AccessToken accessToken, Duration refreshMargin, Duration expirationMargin) {
     super(accessToken, refreshMargin, expirationMargin);
     this.quotaProjectId = null;
+    this.universeDomain = null;
+  }
+
+  /**
+   * A helper for overriding the toString() method. This allows inheritance of super class fields.
+   * Extending classes can override this implementation and call super implementation and add more
+   * fields. Same cannot be done with overriding the toString() directly.
+   *
+   * @return an instance of the ToStringHelper that has public fields added
+   */
+  protected ToStringHelper toStringHelper() {
+    return MoreObjects.toStringHelper(this)
+        .omitNullValues()
+        .add("quotaProjectId", quotaProjectId)
+        .add("universeDomain", getUniverseDomain());
+  }
+
+  @Override
+  public String toString() {
+    return toStringHelper().toString();
+  }
+
+  @Override
+  public boolean equals(Object obj) {
```

Comment:
nit: we should either include access token here too or not add it the authorized user credentials I think.

Reply:
We don't want to do that. I did include it to Credentials that historically has it. Its just two. But in general they don't want to include access token